urwid/display/_raw_display_base.py

```diff
@@ -682,25 +682,25 @@ def using_standout_or_underline(a):
                     icss = escape.IBMPC_ON
                 else:
                     icss = escape.SO
-                o += ["\x08" * back, ias, icss, escape.INSERT_ON, inserttext, escape.INSERT_OFF]
+                output += ["\x08" * back, ias, icss, escape.INSERT_ON, inserttext, escape.INSERT_OFF]
 
                 if cs == "U":
-                    o.append(escape.IBMPC_OFF)
+                    output.append(escape.IBMPC_OFF)
             if whitespace_at_end:
-                o.append(escape.ERASE_IN_LINE_RIGHT)
+                output.append(escape.ERASE_IN_LINE_RIGHT)
 
         if r.cursor is not None:
             x, y = r.cursor
-            o += [set_cursor_position(x, y), escape.SHOW_CURSOR]
+            output += [set_cursor_position(x, y), escape.SHOW_CURSOR]
             self._cy = y
 
         if self._resized:
             # handle resize before trying to draw screen
             return
         try:
-            for line in o:
+            for line in output:
                 if isinstance(line, bytes):
-                    line = line.decode("utf-8", "replace")  # noqa: PLW2901
```

Comment:
this buggy line screams: we should review `Canvas` logic and ultimately switch to the Unicode strings in canvas contents.
Currently we have standard chain for 99,9% of widgets str -> bytes -> str (terminal encoding independent).
During transcoding we're using "target encoding", but in fact it useful only in input processing (not everywhere: in windows and with curses we already can have decoded data).